src/services/NotificationService.ts

```diff
@@ -21,30 +18,23 @@ class NotificationService {
    * Sending email with the certificate according to the given params
    * @param params - personalization details,email and certificate
    * @param emails - emails to send to
-   * @param emailType - email receiver type
+   * @param activityId - activityId for logging purposes
    */
-  public async sendNotification(params: any, emails: string[], emailType: string, activityId: string): Promise<any[]> {
+  public async sendNotification(params: any, emails: string[], activityId: string) {
     const templateId: string = await this.config.getTemplateIdFromEV();
     const emailDetails = {
       personalisation: params,
     };
-    const sendEmailPromise = [];
 
     for (const email of emails) {
-      const sendEmail = this.notifyClient.sendEmail(templateId, email, emailDetails).then((response: any) => response.data);
-      sendEmailPromise.push(sendEmail);
+      try {
```

Comment:
Hmm, I'm not sure. I think it's about the types of fail, if this fails, I think it should '500' rather than log and move on, so it not being managed inside the catch means it will propagate to the handler which does throw. Thoughts?

Reply:
Maybe the templateId should be retrieved prior to calling this method, and be passed through assuming it has been retrieved without issue. Really, we shouldn't be attempting to send any notifications with a templateId that we don't know for sure is not undefined?

Reply:
You do raise a fair point regarding the type of fail.

Reply:
I think leaving it where it is is okay personally. It's a key part of sending a message so should be inside the method, and now this is only called once if we moved it outside we don't really gain a benefit, it will just be moved outside then passed directly in